src/vs/workbench/browser/editor.ts

```diff
@@ -93,6 +103,10 @@ class EditorRegistry implements IEditorRegistry {
 			const matchingDescriptors: EditorDescriptor[] = [];
 
 			for (const editor of this.editors) {
+				if (!editor.isPreferredEditorForResource(input.getResource())) {
+					continue;
+				}
```

Comment:
@mjbvz this is pretty deadly 💀 for all other editors: since no other editor implements this you cannot open "normal" editors anymore. @bpasero will know best but I am not sure about bringing the concept of preferred editors down to the service of if this should be something for the open/open-with commands?

Reply:
Thanks. Yes this change can be deleted. It was from an earlier approach but the current prototype works fine without it